src/components/Footer.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+import { FilterTodo } from './FilterTodo';
+
+interface Props {
+  todos: Todo[],
+  filter: Filter,
+  setFilter: (filter: Filter) => void;
+}
+
+export const Footer: React.FC<Props> = ({ todos, filter, setFilter }) => (
+  <footer className="todoapp__footer" data-cy="Footer">
+    <span className="todo-count" data-cy="todosCounter">
+      {`${todos.length} items left`}
+    </span>
+
+    <nav className="filter" data-cy="Filter">
+      <FilterTodo
+        filter={filter}
+        setFilter={setFilter}
+        filterType={Filter.All}
+        filterLink="FilterLinkAll"
```

Comment:
let's add an enum for data-cy too if you want to implement the component this way
```suggestion
        dataCy="DataCy.FilterLinkAll"
```